### ISSUE TITLE
[Suggestion] Split out each library

### ISSUE THREAD

**Issue by rwu823** [almende/vis#1149]
Currently in `dist` folder it have full size version of vis only, but they are too large not conducive to load at the first time for the users and mostly we may just use one of them.

So I think split out each library is better.

Just like:
```
dist/
  vis.network.js
  vis.timeline.js
  vis.graph2d.js
  vis.graph3d.js
  vis.all.js
```

**Comment by AlexDM0** [comment]
Hi,

This has been mentioned before and we plan to do this eventually. I'm not sure when we'll get around to this though. 

Regards

**Comment by allayli** [comment]
You can build a customized one, by following the procedure: https://github.com/almende/vis#build

**Comment by rwu823** [comment]
@allayli 
Thanks the information. but the mostly case we just 
`npm install --save vis` then play it.

I think user don't need spent other extra time for how to build is better.

**Comment by AlexDM0** [comment]
Hi,

If we split it out as you suggest, npm install will install all of those. We don't plan to make multiple npm packs so you could do npm install visnetwork

Regards

**Comment by rwu823** [comment]
Hi

yeah don't need make multiple npm packages and  npm install all of those is well.
Just split it out. then user can free to choose which one was required.

```html
<script src="'vis.network.js"></script>
```
or
```js
require('vis.network')
```

**Comment by allayli** [comment]
Hi,
Rather than splitting, I suggest that use a website for downloading a customize build, like fabricjs: http://fabricjs.com/build/
for the following reasons:
1. each part of visjs is not independent, they share some common code
2. user may want to exclude some libraries, such as  momentjs; or use more than a part of visjs;  there are many combination
3. you can bundle visjs in your application: https://github.com/almende/vis#example-3-bundle-visjs-as-part-of-your-commonjs-application

Regards

**Comment by josdejong** [comment]
@allayli that's indeed cool :). There is one important drawback though with this approach: when people create a custom build like this, they will not easily update to a newer version of the library as they have to redo the custom build by hand.

I think as a first solution we will add versions containing just a single visualization to `./dist`, like `./dist/vis.timeline.js`, `./dist/vis.network.js`, etc. I think that will cover 90% of what people need.

**Comment by josdejong** [comment]
I've just created separate bundles for the individual visualizations, it will be available in the first next release.

**Comment by josdejong** [comment]
This is now available in v4.16.0
